app/src/main/java/com/android/sample/ui/components/NoInternetScreen.kt

```diff
@@ -1,41 +1,87 @@
-import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.PaddingValues
-import androidx.compose.foundation.layout.Spacer
-import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.graphicsLayer
 import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.TextStyle
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import com.android.sample.R
+import com.android.sample.resources.C.Tag.IMAGE_SIZE
 import com.android.sample.resources.C.Tag.LARGE_PADDING
-import com.android.sample.resources.C.Tag.MEDIUM_PADDING
-import com.android.sample.resources.C.Tag.SUBTITLE_FONTSIZE
 import com.android.sample.resources.C.Tag.TITLE_FONTSIZE
 
 @Composable
-fun NoInternetScreen(paddingValues: PaddingValues) {
-  Column(
-      horizontalAlignment = androidx.compose.ui.Alignment.CenterHorizontally,
-      verticalArrangement = androidx.compose.foundation.layout.Arrangement.Center,
-      modifier = Modifier.fillMaxSize().padding(paddingValues)) {
-        Image(
-            painter = painterResource(id = R.drawable.no_signal),
-            contentDescription = R.string.no_internet_connection.toString())
-        Spacer(modifier = Modifier.padding(LARGE_PADDING.dp))
-        Text(
-            text = R.string.no_internet_connection.toString(),
-            style = TextStyle(fontWeight = FontWeight.Bold, fontSize = TITLE_FONTSIZE.sp))
-        Spacer(modifier = Modifier.padding(MEDIUM_PADDING.dp))
-        Text(
-            text = R.string.internet_connection_ask.toString(),
-            style = TextStyle(fontWeight = FontWeight.SemiBold, fontSize = SUBTITLE_FONTSIZE.sp))
+fun NoInternetScreen() {
+  val isSpinning = true
+  val spinningDuration = 1500
+  val fadingDuration = 1000
+  Box(
+      modifier = Modifier.fillMaxSize().background(Color.Black.copy(alpha = 0.5f)),
+      contentAlignment = Alignment.Center) {
+        val logo = painterResource(id = R.drawable.aptivity)
+        if (isSpinning) {
+          val infiniteTransition = rememberInfiniteTransition(label = "")
+
+          // Animation for rotation
+          val rotation by
+              infiniteTransition.animateFloat(
+                  initialValue = 0f,
+                  targetValue = 360f,
+                  animationSpec =
+                      infiniteRepeatable(
+                          animation =
+                              tween(durationMillis = spinningDuration, easing = LinearEasing)),
```

Comment:
Would be good to replace spinningDuration by a constant placed in C

Reply:
t was done in purpose, I don't think that we will need this and adding this in constants would just make it more confusing